system/security/shared/caching.cpp

```diff
@@ -679,11 +698,102 @@ bool CPermissionsCache::queryPermsManagedFileScope(ISecUser& sec_user, const cha
     return rc;
 }
 
-SecAccessFlags CPermissionsCache::queryDefaultPermission(ISecUser& user)
+
+void CPermissionsCache::managedFileScopesCacheFillThread()
+{
+    DBGLOG("CACHE: CPermissionsCache managedFileScopesCacheFillThread starting");
+
+    // Signal anyone waiting that the fill thread has started
+    m_fileScopeCacheFillThreadReadySem.signal();
+
+    unsigned waitTimeSeconds = m_cacheTimeoutInSeconds;
+    while (true)
+    {
+        m_fileScopeCacheFillSem.wait(waitTimeSeconds * 1000);
+
+        // If either explicitly stopping the thread or the cache has been disabled, exit
+        if (m_stopFileScopeCacheFillThread || m_cacheTimeoutInSeconds == 0)
+            break;
+
+        time_t now;
+        time(&now);
+
+        time_t copyLastCacheFillTime = m_lastCacheFillTime.load();
+
+        // If not filled since we just got the time
+        if (copyLastCacheFillTime < now)
```

Comment:
This test is almost impossible to hit, and confuses the code.  Why is it here?

Reply:
If this test was deleted and the "impossible" condition happened, the only consequence would be if
-ve-number timeout - which would correctly not succeed.
(time_t is guaranteed to be signed.)

However it isn't quite so simple ...  m_cacheTimeoutInSeconds is defined as an unsigned, so the comparison could be signed - if time_t is the same width as unsigned.  That would be true on very old systems with 32bit time_t.

Safest is for m_cacheTimeoutInSeconds to also be a time_t, so there is no ambiguous mixing of signed and unsigned arguements.